academic_helper/logic/schedule.py

```diff
@@ -40,3 +41,12 @@ def get_all_classes(course_number: str) -> List[dict]:
         classes = CourseClass.objects.filter(group_id=group["id"])
         group["classes"] = [c.as_dict for c in classes]
     return serialized
+
+
+def get_courses_by_groups(groups: List[int]):
+    occurrences_ids = ClassGroup.objects.filter(id__in=groups).values_list('occurrence', flat=True)
+    courses_ids = CourseOccurrence.objects.filter(id__in=occurrences_ids).values_list('course', flat=True)
+    courses = Course.objects.filter(id__in=courses_ids)
+    serialized = [c.as_dict for c in courses]
```

Comment:
I think this can be done shorter with `select_related`, but its not critical at all.